### PR TITLE
[CORDA-446] Remove entities from netmap cache & friends

### PR DESCRIPTION
**Pull request by al-r3**
Remove NetworkMapService and all implementations.
They are not used in the new NetworkMap implementation

### CODE REVIEW

File: core/src/test/kotlin/net/corda/core/flows/ContractUpgradeFlowTest.kt [hunk 1]
```diff
@@ -127,7 +128,8 @@ class ContractUpgradeFlowTest {
 
     @Test
     fun `2 parties contract upgrade using RPC`() {
-        rpcDriver(initialiseSerialization = false) {
+        rpcDriver(initialiseSerialization = false,
+                portAllocation = PortAllocation.RandomFree) {
```
Comment (shamsasari) [inline core/src/test/kotlin/net/corda/core/flows/ContractUpgradeFlowTest.kt]: You need to undo this?

File: node/src/main/kotlin/net/corda/node/services/network/NetworkMapService.kt [hunk 2]
```diff
@@ -1,16 +1,5 @@
 package net.corda.node.services.network
```
Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/services/network/NetworkMapService.kt]: This file doesn't have anything in it.

File: testing/node-driver/src/main/kotlin/net/corda/testing/driver/Driver.kt [hunk 3]
```diff
@@ -492,10 +491,10 @@ fun addressMustBeBoundFuture(executorService: ScheduledExecutorService, hostAndP
 
 /*
  * The default timeout value of 40 seconds have been chosen based on previous node shutdown time estimate.
- * It's been observed that nodes can take up to 30 seconds to shut down, so just to stay on the safe side the 40 seconds
+ * It's been observed that nodes can take up to 30 seconds to shut down, so just to stay on the safe side the 60 seconds
  * timeout has been chosen.
  */
-fun addressMustNotBeBound(executorService: ScheduledExecutorService, hostAndPort: NetworkHostAndPort, timeout: Duration = 40.seconds) {
+fun addressMustNotBeBound(executorService: ScheduledExecutorService, hostAndPort: NetworkHostAndPort, timeout: Duration = 60.seconds) {
```
Comment (shamsasari) [inline testing/node-driver/src/main/kotlin/net/corda/testing/driver/Driver.kt]: revert

File: node/src/main/kotlin/net/corda/node/services/network/NetworkMapService.kt (hunk 2 above)
Comment (al-r3) [reply node/src/main/kotlin/net/corda/node/services/network/NetworkMapService.kt]: fair point :)

I've just deleted all the comments, they seems significantly outdated.

**Comment by shamsasari** [comment]
Also remove the minimumPlatformVersion config. It's only used when the node was a network map

**Comment by al-r3** [comment]
The minimumPlatformVersion is used at the startup in AbstractNode to compare the version in the config with the one in the manifest.
I think that sill makes sense (and is otherwise unrelated with the network map node)

**Comment by shamsasari** [comment]
But that serves no purpose on its own. That config was the first network parameter and is now no longer relevant. I'll remove it my PR.

**Pull request merged**